ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -94,6 +96,59 @@ func ConvertToTaskDefinition(params ConvertTaskDefParams) (*ecs.TaskDefinition,
 		return nil, err
 	}
 
+	executionRoleArn := taskDefParams.executionRoleArn
+
+	// Check for and apply provided ecs-registry-creds values
+	if params.PrivRegistryContext != nil {
```

Comment:
Could all of this be done in a stand alone function?

Reply:
So, my reasoning for keeping it here was that I'm modifying 2 diff top-level task def fields (container defs and task execution role). I did consider extracting the container def stuff into it's own function, but didn't like the idea of passing around or recreating the container defs after `convertToContainerDef` was called.

However, I suppose this could be accomplished by passing in a pointer to the container defs (and updating the values in place) or doing this processing as part of `convertToContainerDef`; however the latter means the execution role & cred param portion of the `PrivRegistryContext` would be processed in 2 diff places which I'm not a fan of. Thoughts on either of those approaches?

Reply:
I think I'd favor just replacing this code block with a function call that does the same stuff, and takes pointers as arguments where needed.